src/pages/Staking/index.tsx

```diff
@@ -168,6 +167,27 @@ const Staking = (): JSX.Element => {
   );
   useErrorHandler(voteGovernanceTokenBalanceError);
 
+  // My currently claimable rewards
+  const {
+    isIdle: currentRewardAmountIdle,
+    isLoading: currentRewardAmountLoading,
+    data: currentRewardAmount,
+    error: currentRewardAmountError,
+    refetch: currentRewardAmountRefetch
+  } = useQuery<MonetaryAmount<Currency<GovernanceUnit>, GovernanceUnit>, Error>(
+    [
+      GENERIC_FETCHER,
+      'escrow',
+      'getRewards',
+      address
+    ],
+    genericFetcher<MonetaryAmount<Currency<GovernanceUnit>, GovernanceUnit>>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(currentRewardAmountError);
```

Comment:
@nud3l 
I think we should move this into `ClaimRewardsButton` component because `getRewards` API does not have to do with other APIs (only used for `ClaimRewardsButton` component). Besides that, staking and claiming actions are sort of independent of each other.

Reply:
Claimable rewards are changed depening on the staked amount and time so this should be here IMO

Reply:
As you can see in the react-query, the only argument is the user's account address. The staked amount and time affect it for sure but they are reflected not in this form but internally on the parachain level. So simply speaking, the form environment (like input locking amount and time) does not affect this claimable rewards.

In that case, we should nest the API call inside the relevant component and the fetched data is also only consumed in that relevant component.

Let me try to take some examples of the disadvantages:
- If `getRewards` API fails for any reason, the whole `Staking` component is affected although we could limit that affection to `ClaimRewardsButton` which would be more healthy and maintainable.
- As you can see, the code of `Staking` is very lengthy which makes the code less readable, we should distribute the code as much as possible by moving component-specific parts into the relevant components. It's one of the best practices.
- `claimableRewardAmountIdle` and `claimableRewardAmountLoading` are added to `initializing` (loading procedure) so the staking UI is to be waiting for that period of time until it's fully functionable.
Actually, we also can just remove them here as well and we should because if it takes quite a long time (e.g. let's assume it takes one hour in order to make things more noticeable) to load `getRewards` for any reason, the user will not be able to take staking actions at the moment due to this `getRewards` API call which does not directly have to with staking actions.